source/code/plugin/out_mdm.rb

```diff
@@ -12,27 +12,36 @@ def initialize
       require "net/http"
       require "net/https"
       require "uri"
-      require 'yajl/json_gem'
+      require "yajl/json_gem"
       require_relative "KubernetesApiClient"
       require_relative "ApplicationInsightsUtility"
 
       @@token_resource_url = "https://monitoring.azure.com/"
       @@grant_type = "client_credentials"
       @@azure_json_path = "/etc/kubernetes/host/azure.json"
       @@post_request_url_template = "https://%{aks_region}.monitoring.azure.com%{aks_resource_id}/metrics"
-      @@token_url_template = "https://login.microsoftonline.com/%{tenant_id}/oauth2/token"
+      @@aad_token_url_template = "https://login.microsoftonline.com/%{tenant_id}/oauth2/token"
+
+      # msiEndpoint is the well known endpoint for getting MSI authentications tokens
+      @@msi_endpoint_template = "http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&client_id=%{user_assigned_client_id}&resource=%{resource}"
```

Comment:
egress needs to be opened up ?

Reply:
when I did the spike, this was accessible out of the box with standard ports.

Reply:
checking with Saurya regarding this